### PR TITLE
issue 521.1: do not mark loadPlan as valid unless all instances were created successfully

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3144/

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -462,71 +466,97 @@ void ClusterInfo::loadPlan() {
...
-        for (auto const& database : VPackObjectIterator(databasesSlice)) {
-          try {
-            name = database.key.copyString();
-          } catch (arangodb::velocypack::Exception const& e) {
-            LOG_TOPIC(ERR, Logger::AGENCY)
-                << "Failed to get database name from json, error '" << e.what()
-                << "'. VelocyPack: " << database.key.toJson();
-            throw;
-          }
+  auto slice = result.slice()[0].get( // get slice
```
Comment (gnusi) [inline arangod/Cluster/ClusterInfo.cpp]: should we check if `result().slice()` is array?

File: arangod/Cluster/ClusterInfo.cpp [hunk 2]
```diff
@@ -912,128 +957,148 @@ void ClusterInfo::loadCurrent() {
...
+      << " httpCode: " << result.httpCode() << " errorCode: " << result.errorCode()
+      << " errorMessage: " << result.errorMessage() << " body: " << result.body();
 
-    auto currentBuilder = std::make_shared<VPackBuilder>();
-    currentBuilder->add(slice);
+    return;
+  }
 
-    VPackSlice currentSlice = currentBuilder->slice();
+  auto slice = result.slice()[0].get( // get slice
```
Comment (gnusi) [inline arangod/Cluster/ClusterInfo.cpp]: should we check if `result.slice()` is array?

File: arangod/Cluster/ClusterInfo.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/Cluster/ClusterInfo.cpp]: done

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3155/

File: arangod/Cluster/ClusterInfo.cpp (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/Cluster/ClusterInfo.cpp]: done

**Pull request merged**